code/modules/vending/vending.dm

```diff
@@ -1586,7 +1616,335 @@
+		else
+			html_parts += "<a href='?src=\ref[src];unlock=true'>[src.cardname] "
+			if (!unlocked == 1) html_parts += "(locked) </a></br>"
+			else html_parts += "(unlocked) </a></br>"
+		html_parts += "Loading Chute:  "
+		if(!loading == 0)
+			html_parts += "<a href='?src=\ref[src];loading=false'>Open</a></br> "
+		else
+			html_parts += "<a href='?src=\ref[src];loading=true'>Closed</a></br> "
+		html_parts += "Vendor Name:  "
+		html_parts += "<a href='?src=\ref[src];rename=true'>[src.name]</a> "
+		html_parts += "</small></td></tr></tbody></table></TT><br>"
+		src.wire_HTML += jointext(html_parts, "")
+
+	attackby(obj/item/target, mob/user)
+		if(loading && panel_open)
+			addProduct(target, user)
+		else
+			. = ..()
+		if(!panel_open) //lock up if the service panel is closed
+			loading = 0
+			unlocked = 0
+		src.generate_HTML(1)
+	//Save and restore icon state in case we're in product display mode
+	power_change()
+		var/original_icon = src.icon_state
+		. = ..()
+		if (!(status & BROKEN) && powered())
+			src.icon_state = original_icon
+			setCrtOverlayStatus(1)
+		else
+			setCrtOverlayStatus(0)
+
+	Topic(href, href_list)
+		. = ..()
+		if (status & (BROKEN|NOPOWER))
+			setCrtOverlayStatus(0)
+			return
+		if (usr.stat || usr.restrained() || !in_interact_range(src, usr))
+			return
 
+		if (href_list["loading"])
+			if (src.panel_open && src.unlocked)
+				loading = !loading
+				src.generate_HTML(0, 1)
+		else if (href_list["unlock"] && src.panel_open)
+			if (!owner && src.scan?.registered)
+				owneraccount = FindBankAccountByName(src.scan.registered)
+				owner = src.scan.registered
+				cardname = src.scan.name
+				unlocked = 1
+				loading = 1
+			else if (src.scan?.registered && owner == src.scan.registered)
+				unlocked = !unlocked
+				if(!unlocked && loading) loading = 0
+			else
+				unlocked = 0
+				loading = 0
+			src.generate_HTML(1, 1)
+		else if (href_list["rename"] && src.panel_open && src.unlocked)
+			var/inp
+			inp = html_encode(sanitize(input(usr,"Enter new name:","Vendor Name", "") as text))
+			if(inp && inp != "" && (usr.stat || usr.restrained() || in_interact_range(src, usr)))
+				src.name = inp
+				src.generate_HTML(0, 1)
+				generate_slogans()
+		else if (href_list["setprice"] && src.panel_open && src.unlocked)
+			var/inp
+			inp = input(usr,"Enter the new price:","Item Price", "") as num
+			if(inp && (usr.stat || usr.restrained() || in_interact_range(src, usr)))
+				var/datum/data/vending_product/player_product/R = locate(href_list["setprice"]) in src.player_list
+				R.product_cost = inp
+				src.generate_HTML(1, 0)
+		else if(href_list["vend"] && (length(player_list) <= 0))
+			src.setItemOverlay(null)
+			icon_state = "player"
+		else if(href_list["icon"] && src.panel_open && src.unlocked)
+			var/datum/data/vending_product/player_product/R = locate(href_list["icon"]) in src.player_list
+			src.setItemOverlay(R.icon)
+			src.generate_HTML(1, 0)
```

Comment:
![image](https://user-images.githubusercontent.com/1017374/115107762-b6877580-9f21-11eb-9362-84860b6c3af0.png)
Overlays seem to get wonky.... Can only seem to update image if I stick something in and only to the thing I put in?

![image](https://user-images.githubusercontent.com/1017374/115108420-38c56900-9f25-11eb-9222-a7ad5ffb701c.png)
![image](https://user-images.githubusercontent.com/1017374/115108491-9e195a00-9f25-11eb-9768-9b825069eced.png)


Consider a general function to update all the overlays to things don't get out of sync.

Reply:
>Overlays seem to get wonky.... Can only seem to update image if I stick something in and only to the thing I put in?

This is the only functionality available in this PR, in the future you'll be able to choose from a selection of promotional images.

Reply:
Then only show anchor astrix for items that will work, otherwise it feels broken.

Unable to clear loading chute, always depicted.

Reply:
Completely impossible, the asterisk is there so you can choose ones that work.
I don't know anything about product sprites, I have 32x32 and I shrink them to a size that lets the widest items fit.
This is all that can be done.

Reply:
I'll resolve this cause anything else will be about the manager